### ISSUE TITLE
mouse cursor is not align with bounding boxes of main menu text

### ISSUE THREAD

**Issue by palmerj** [FWGS/xash3d-fwgs#921]
When I try to select the menu options with the mouse cursor, the alignment between the bounding boxes of menu text and mouse cursor position is offset. I can't select more than the top couple of menu items by trying to scroll to the bottom of the window. Using the keyboard does, however, work ok. See screenshot

<img width="1602" alt="menu" src="https://user-images.githubusercontent.com/580022/177725716-669044d5-2862-4f4c-9302-deb6ed77ba7e.png">

I'm starting the game like this:

```
XASH3D_EXTRAS_PAK1=extras.pak ./xash3d -windowed -width 1600 -height 1200
```

I've also tested in fullscreen mode, and it's the same result. 

My desktop resolution is Resolution: 2880 x 1800 Retina and scaling is on so it looks like 1920x1200. If I turn scaling off (i.e no high DPI), it's still the same result. 

So far the rest of the game seems to run fine.

Note I've compiled this on MacOS 12.4 64bit (32bit was removed from MacOS long ago). I'm not sure if this is part of the issue, as I can't compile the VGUI because I'm on MacOS and 64bit...

I'm happy to try hacking the game to see if I can find a fix. I just need someone to let me know which source files I should start looking in. 

Thanks heaps.

**Comment by palmerj** [comment]
> This still compiles for Mac? We dropped support for a while ago...

It does indeed. I saw your issue to drop support, so was pretty happy when it compiled!

> If you have programming skills, you can try to debug it, by looking into IN_MouseMove function in engine/client/input.c. Every frame it requests mouse position from SDL through Platform_GetMousePos and then propagates it to VGUI and MainUI.

Thanks, I will take a look. So in the [waf configure process](https://gist.github.com/palmerj/a983982d39403935ca9282dd1510e46c) it mentions:

```
--> vgui_support                                        : in progress
Does this architecture support VGUI?                    : no
```

What does that mean? Seems like vgui isn't supported. Is that the issue?

**Comment by lewa-j** [comment]
We do not use vgui for main menu. So it's unrelated.

**Comment by nekonomicon** [comment]
Vgui needed only for command menu and bhop scripts with `_special` cmd(which broken in steam).

**Comment by palmerj** [comment]
> It's developer console, you need to run engine with -dev 2 or -console flag to show it.

Ah ok, sorry my misunderstanding. This is great news then

So I debugged things a little and found the best solution is just setting `vid_highdpi "0"` within video.cfg. I searched around in the code a little and found that multiple changes would be needed to make the text and cursor rectangle checks work well. Likely would need to also use the SDL_GL_GetDrawableSize and SDL_GetWindowSize SDL2 APIs to get the HiDPI scale factor and then apply scaling that to the menu items.

**Comment by a1batross** [comment]
In my opinion, operating system or window compositor should not scale games because then it gets blurry. For example, there is a specific hack for Windows to disable scaling (DPI-awareness). Yes, it may look smaller, but then we also make use of every pixel.

**Comment by palmerj** [comment]
Thanks @a1batross. Are you saying you would still like to keep this open to fix the behaviour rather than use the vid_highdpi "0" setting?

**Comment by a1batross** [comment]
No, but we probably have to remove ALLOW_HIGH_DRI flag (which makes vid_highdpi completely unnecessary)

It looks like I've mistaken when I thought this flag removes scaling.

**Comment by a1batross** [comment]
Fixed in 7157c3b4
